viennacl/src/main/scala/org/apache/mahout/viennacl/opencl/javacpp/DenseColumnMatrix.scala

```diff
@@ -36,12 +36,24 @@ import org.bytedeco.javacpp.annotation._
 @Name(Array("viennacl::matrix<double,viennacl::column_major>"))
 final class DenseColumnMatrix(initDefault:Boolean = true) extends MatrixBase {
 
-  def this(nrow: Int, ncol: Int, ctx: Context = new Context()) {
+  def this(nrow: Int, ncol: Int) {
+    this(false)
+    allocate(nrow, ncol, new Context())
+  }
+
+  def this(nrow: Int, ncol: Int, ctx: Context) {
     this(false)
     allocate(nrow, ncol, ctx)
   }
 
-  def this(data: DoublePointer, nrow: Int, ncol: Int, ctx: Context = new Context(Context.MAIN_MEMORY)) {
```

Comment:
@rawkintrevo I have no problems building this for scala-2.11, with E.g.: `mvn clean install -Dscala.version=2.11.2 -Dscala.compat.version=2.11 -Pviennacl -Phadoop2  -DskipTests`` *note* the scala version is @ 2.11 which I think wee may need to make our cap due to the Breeze vector cap @scala 2.11.2 :
```
 [WARNING]  org.scalanlp:breeze_2.11:0.11.2 requires scala version: 2.11.4 
```
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Mahout Build Tools ................................. SUCCESS [  3.189 s]
[INFO] Apache Mahout ...................................... SUCCESS [  0.080 s]
[INFO] Mahout Math ........................................ SUCCESS [ 13.226 s]
[INFO] Mahout HDFS ........................................ SUCCESS [  3.310 s]
[INFO] Mahout Map-Reduce .................................. SUCCESS [ 21.442 s]
[INFO] Mahout Integration ................................. SUCCESS [  6.183 s]
[INFO] Mahout Examples .................................... SUCCESS [ 21.106 s]
[INFO] Mahout Math Scala bindings ......................... SUCCESS [ 54.202 s]
[INFO] Mahout Spark bindings .............................. SUCCESS [ 51.939 s]
[INFO] Mahout Native VienniaCL OpenCL Bindings ............ SUCCESS [ 48.965 s]
[INFO] Mahout Native VienniaCL OpenMP Bindings ............ SUCCESS [ 25.782 s]
[INFO] Mahout Release Package ............................. SUCCESS [  2.567 s]
[INFO] Mahout H2O backend ................................. SUCCESS [ 22.381 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:34 min
[INFO] Finished at: 2017-06-23T16:39:21-07:00
[INFO] Final Memory: 132M/1124M
[INFO] -----------------------------------
```

tests are failing for me though in `viennacl`.  I'll to dig deeper.